### PR TITLE
Fix download of fixed assets (on top of Adam's PR)

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/File.pm [hunk 1]
```diff
@@ -76,10 +76,17 @@ sub test_file {
     return $self->serve_static_($self->param('filename'));
 }
 
+sub download_asset {
+    my ($self) = @_;
+    # we handle this in apache
+    return $self->reply->not_found;
```
Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/File.pm]: I know, that apache is required right now, but I always keep in mind that this requirement should be "fixed". Given this, should I consider apache/nginx/... as mandatory for the future and stop worrying about it?

Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/File.pm]: For now this route is mainly documenting the requirement.

Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/File.pm]: (and the route is not covered in tests)

**Comment by AdamWill** [comment]
Wow. It's your project, so it's up to you, but I'm amazed that you think this is *better*, because it does exactly what `locate_asset` was written to avoid: spreads the logic about what kinds of assets we have and where they might be found into two other places.

Imagine we invent another type of asset, or change the way `fixed` assets are implemented. With my approach, for this code, we then fix *one* place: `locate_asset`. With your code, we have to fix *three* places (and maybe change an interface, that `download_fixed_asset` route name).

The reason this change was annoying to write in the first place, and the reason the bug this patch fixes happened in the first place, is because instead of having a single function for 'find this asset based on its type' we had dozens of placing reimplementing 'combine the asset type and name with a known location', instead of a single function doing it. So I wrote one...but now you just want to ignore it and have things do the work themselves again? Well, okay.

**Comment by coolo** [comment]
Well, right now we have only place that actually maps the file path to a URL. If we get a second, we should move this into a function. But I wanted to avoid doing this when there is no controller. Because your way has drawbacks too - e.g. if you mount the app in a suburl.

**Comment by AdamWill** [comment]
Sure, I see the point, but we can still do better than that ugly regex match :) I have some ideas on this, I'll try and find time to send a PR on Monday...it just would've been nicer to kick this around a bit rather than simply finding this merged without any opportunity to comment on it :/

**Comment by coolo** [comment]
we can have an url that is just /assets/*assetpath

**Comment by coolo** [comment]
https://github.com/os-autoinst/openQA/pull/986 - wdyt?

**Pull request merged**